### ISSUE TITLE
Support for Network Access Control in SignalR

### ISSUE THREAD

**Issue by jayendranarumugam** [terraform-providers/terraform-provider-azurerm#11777]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

* azurerm_signalr_service

### Potential Terraform Configuration

```hcl
resource "azurerm_signalr_service" "example" {
  name                = "tfex-signalr"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  sku {
    name     = "Free_F1"
    capacity = 1
  }

  cors {
    allowed_origins = ["http://www.example.com"]
  }

  features {
    flag  = "ServiceMode"
    value = "Default"
  }

network_rules = [
    {
      name      = "AllowClientConnection"
      rule_type = "allow"
      endpoint  = "public-network"
      services  = ["ClientConnection"]
    },
    {
      name      = "DenyAllOthers"
      rule_type = "deny"
      endpoint  = "public-network"
      services  = ["ServerConnection", "RESTAPI"]
    }
  ]

}
```

Corresponds to the image below
![image](https://user-images.githubusercontent.com/25414541/118783280-8cbcc800-b8ac-11eb-85a9-f02b5591ce72.png)

### References

* https://github.com/claranet/terraform-azurerm-signalr

**Comment by jayendranarumugam** [comment]
@katbyte , you can close this issue as with this new https://github.com/terraform-providers/terraform-provider-azurerm/pull/12434, this feature is completed. Thanks, @neil-yechenwei !


For others, you can use the below code to implement NAC. Here is the [official docs](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/signalr_service_network_acl)


```hcl
resource "azurerm_private_endpoint" "example" {
  name                = "example-privateendpoint"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  subnet_id           = azurerm_subnet.example.id

  private_service_connection {
    name                           = "psc-sig-test"
    is_manual_connection           = false
    private_connection_resource_id = azurerm_signalr_service.example.id
    subresource_names              = ["signalr"]
  }
}

resource "azurerm_signalr_service_network_acl" "example" {
  signalr_service_id = azurerm_signalr_service.example.id
  default_action     = "Deny"

  public_network {
    allowed_request_types = ["ClientConnection"]
  }

  private_endpoint {
    id                    = azurerm_private_endpoint.example.id
    allowed_request_types = ["ServerConnection"]
  }
}
```

**Comment by tombuildsstuff** [comment]
Fixed via #12434
